mllib/src/main/scala/org/apache/spark/ml/regression/AFTSurvivalRegression.scala

```diff
@@ -0,0 +1,400 @@
+   * @group param
+   */
+  final val quantile: Param[Vector] = new Param[Vector](this,
+    "quantileCol", "quantile column name")
+
+  /** @group getParam */
+  final def getQuantile: Vector = $(quantile)
+
+  /** Checks whether the input has quantile vector. */
+  protected[ml] def hasQuantile: Boolean = {
+    isDefined(quantile) && $(quantile).size != 0
+  }
+
+  /**
+   * Validates and transforms the input schema with the provided param map.
+   * @param schema input schema
+   * @param fitting whether this is in fitting or prediction
+   * @return output schema
+   */
+  protected def validateAndTransformSchema(
+      schema: StructType,
+      fitting: Boolean): StructType = {
+    SchemaUtils.checkColumnType(schema, $(featuresCol), new VectorUDT)
+    if (fitting) {
+      SchemaUtils.checkColumnType(schema, $(censorCol), DoubleType)
+      SchemaUtils.checkColumnType(schema, $(labelCol), DoubleType)
+    }
+    SchemaUtils.appendColumn(schema, $(predictionCol), DoubleType)
+  }
+}
+
+/**
+ * :: Experimental ::
+ * Fit a parametric survival regression model named Accelerated failure time model
+ * ([[https://en.wikipedia.org/wiki/Accelerated_failure_time_model]])
+ * based on the Weibull distribution of the survival time.
+ */
+@Experimental
+class AFTSurvivalRegression(override val uid: String)
+  extends Estimator[AFTSurvivalRegressionModel] with AFTSurvivalRegressionParams with Logging {
+
+  def this() = this(Identifiable.randomUID("aftReg"))
+
+  /** @group setParam */
+  def setFeaturesCol(value: String): this.type = set(featuresCol, value)
+
+  /** @group setParam */
+  def setLabelCol(value: String): this.type = set(labelCol, value)
+
+  /** @group setParam */
+  def setCensorCol(value: String): this.type = set(censorCol, value)
+  setDefault(censorCol -> "censored")
+
+  /** @group setParam */
+  def setPredictionCol(value: String): this.type = set(predictionCol, value)
+
+  /**
+   * Set if we should fit the intercept
+   * Default is true.
+   * @group setParam
+   */
+  def setFitIntercept(value: Boolean): this.type = set(fitIntercept, value)
+  setDefault(fitIntercept -> true)
+
+  /**
+   * Set the maximum number of iterations.
+   * Default is 100.
+   * @group setParam
+   */
+  def setMaxIter(value: Int): this.type = set(maxIter, value)
+  setDefault(maxIter -> 100)
+
+  /**
+   * Set the convergence tolerance of iterations.
+   * Smaller value will lead to higher accuracy with the cost of more iterations.
+   * Default is 1E-6.
+   * @group setParam
+   */
+  def setTol(value: Double): this.type = set(tol, value)
+  setDefault(tol -> 1E-6)
+
+  /**
+   * Extracts (features, label, censored) from input dataset.
+   */
+  protected[ml] def extractCensoredLabeledPoints(
+      dataset: DataFrame): RDD[(Vector, Double, Double)] = {
+    dataset.select($(featuresCol), $(labelCol), $(censorCol))
+      .map { case Row(features: Vector, label: Double, censored: Double) =>
+      (features, label, censored)
```

Comment:
* We can have a private case class to represent an instance. It makes the code simpler to read.
* fix indentation